woocommerce/wc-admin
# Regenerate report tables using Action Scheduler

Fixes #860.

This PR supplants work done in https://github.com/woocommerce/wc-admin/pull/1050.

This PR introduces (re)generating report tables in batches. The batch size can be customized based on the type (customer, order, etc). In order to avoid timeouts and memory issues when creating many batches (for large databases), the batch creation itself is batched (and that size can be customized).

This PR also introduces a mechanism that allows for job to be delayed until another one is finished. This is used to ensure that the orders and products lookup tables are (re)generated after the customers lookup is finished.

_I tested this against a store with ~6500 customers and ~19000 orders._

**To test:**
- Verify that on install, reports tables are generated (can check Tools > Scheduled Actions if it's taking a while)
- Verify that WooCommerce > Status > Tools > Regenerate Reports works as it did before
- Test against a large dataset (lots of customers, products, orders) - verify that actions get scheduled
